dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/message/NettyRequestMessage.java

```diff
@@ -196,11 +199,52 @@ else if (host.length() > 1 && '/' == host.charAt(0) && '/' == host.charAt(1) &&
                 }
             }
             parseURLHost(host, start, slash_start);
+            parseURI(host.getBytes(), slash_start);
         }
         // Need to check if Scheme is also verified by Netty coded or add that ourselves
         // Probably need to add check of authority ourselves as well
     }
 
+    /**
+     * Parse the URI information out of the input data, along with any query
+     * information found afterwards. If format errors are found, then an
+     * exception is thrown.
+     *
+     * @param data
+     * @throws IllegalArgumentException
+     */
+    private void parseURI(byte[] data, int start) {
+        // at this point, we're parsing /URI [?querystring]
+        if (start >= data.length) {
+            // PK22096 - default to "/" if not found, should have caught empty
+            // string inputs previously (http://host:port is valid)
+            this.myURIBytes = SLASH;
+            if (tc.isDebugEnabled()) {
+                Tr.debug(tc, "Defaulting to slash since no URI data found");
+            }
+            return;
+        }
+        int uri_end = data.length;
+        for (int i = start; i < data.length; i++) {
+            // look for the query string marker
+            if ('?' == data[i]) {
+                uri_end = i;
+                break;
+            }
+        }
+        // save off the URI
+        if (start == uri_end) {
+            // no uri found
+            throw new IllegalArgumentException("Missing URI: " + GenericUtils.getEnglishString(data));
```

Comment:
Add to  translations issue

Reply:
We use the same message [here](https://github.com/OpenLiberty/open-liberty/blob/c8249ca16f2cb8e01cf5b098402ebedd7e7817f7/dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java#L1181) in Legacy. Should we add it or create a separate issue?

Reply:
Add it here https://github.com/OpenLiberty/open-liberty/issues/31761 we can split it up later